primeqa/components/reranker/colbert_reranker.py

```diff
@@ -45,6 +45,14 @@ class ColBERTReranker(BaseReranker):
             "description": "Path to model",
         },
     )
+    model_type: str = field(
+        default="xlm-roberta",
+        metadata={
+            "name": "Model Type",
+            "api_support": True,
+            "description": "Type of encoder architecture",
+        },
+    )
```

Comment:
@riyazbhat @franzmpub More on this 

This component as well as the ColBERTIndexer and ColBERTReranker are used in the service layer to load a ColBERT dnn/model file.  The service layer has no way of knowing what the  `model_type` so this cannot be required in the interface or set in the ColBERTConfig  in the `__post_init__` method.

Since the default is `xlm-roberta` it will work with the DrDecr model but it will fail on any model that is not xlm-roberta here https://github.com/primeqa/primeqa/blob/e25ece79a07f4c6fc85cb640a8d634b8acac8461/primeqa/ir/dense/colbert_top/colbert/modeling/factory.py#L45C5-L45C5

It is possible to look up the `model_type` in the dnn.  However i noticed for the drdecr model, this is set to `xlm-roberta-base`  not `xlm-roberta'   which will cause the code to fail with 
`NotImplementedError: Model type: xlm-roberta-base is not supported.` here https://github.com/primeqa/primeqa/blob/e25ece79a07f4c6fc85cb640a8d634b8acac8461/primeqa/ir/dense/colbert_top/colbert/modeling/factory.py#L72
`
It seems additional changes are need to fully support loading the models.  The components cannot require `model_type` to be specified.

Reply:
> @riyazbhat (FYI @bhavani105) One possible solution is the map the model type as in the .dnn file to the corresponding model type as supported in the `if` statements starting here: [factory.py](https://github.com/primeqa/primeqa/blob/colbert_encoder_types/primeqa/ir/dense/colbert_top/colbert/modeling/factory.py#L53) This will require just a simple "many to one" mapping structure introduced in `factory.py` mapping e.g. "xlm-roberta-base", "xlm-roberta-large" and other "xlm-roberta-XYZ" flavors to "xlm-roberta". What do you think?

Each model type has a corresponding HF config from which you can get a list of specific types e.g. 
https://github.com/huggingface/transformers/blob/b71f20a7c9f3716d30f6738501559acf863e2c5c/src/transformers/models/xlm_roberta/configuration_xlm_roberta.py#L27

Reply:
@riyazbhat , @bhavani105 : I've made changes in `factory.py` to allow mapping from the model subtypes to the types supported in `factory.py`.  Also made the default `model_type` in `components/reranker/colbert_reranker.py` set to `roberta`, same as in `components/retriever/dense.py`.
This now works for e.g. the DrDecr model, reading "xlm-roberta-base" from the model's config and mapping it to "xlm-roberta".

Reply:
@bhavani105 (FYI @riyazbhat ): I've made changes in `ir/dense/colbert_top/colbert/modeling/factory.py`, `components/reranker/colbert_reranker.py` and `components/retriever/dense.py` to avoid using the `model_type` parameter in the `components` classes.
Verified loading a model both from a checkpoint file and from a directory containing `config.json`.